### DISCUSSION TITLE
Stable derived colors from data

### DISCUSSION THREAD

**Discussion by ankon** [grafana/grafana#38357] [Dashboards, Panels & Visualizations]
**What would you like to be added**:

I would like to see a generic visualization option to pick colors for metrics not randomly, but rather based on some label/field/data item related to the metric.

**Why is this needed**:

I have a lot of graphs which show data separate per "tenant". There are many such tenants, but not all of them have at any point in time a non-null value for the metric. What then happens is that grafana picks colors for the metrics purely on what is rendered, which means I need to always check the legend to map "tenant" to "color". 

What I would like to is have a way to simply say "derive the color from the tenant", for example by hashing the tenant and then mapping that hash value to a color. This way the color would always be stable, and I can quickly glance at the graph and understand the displayed data.

An example query from our dashboard:

```
sum(gateway_connections{kubernetes_namespace=~"$Environment"}) by (kubernetes_namespace,tenant)
```

I would like to use the 'tenant' label as source for the color.

**Comment by ryantxu** [comment]
See also #16912 -- we are making progress on letting options be driven consistently from the datasource (not just the panel)

**Reply by jacksongoode** [reply]
@ryantxu Is this still on the roadmap?

**Reply by leeoniya** [reply]
https://github.com/grafana/grafana/pull/66197

**Comment by pkolyvas** [comment]
We're in the process of moving Grafana feature requests to [GitHub Discussions](https://community.grafana.com/t/were-moving-grafana-feature-requests-to-github-discussions/50341/3). This issue is being moved to [Discussions](https://github.com/grafana/grafana/discussions) or has been moved from our issues list, depending on where you're seeing this message. 

 We hope to use the discussion thread to continue building use cases, and additional evidence based on engagement/voting, in order to help us prioritize this feature request later on down the road. 

As with all things Grafana, we value your input and participation in this project and appreciate your patience during this transitional period. Looking forward to the discussion!

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
